src/3rd_party.c

```diff
@@ -0,0 +1,136 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define _GNU_SOURCE
+#include "3rd_party.h"
+#include "config.h"
+#include "lib/log.h"
+#include "swupd.h"
+#include <getopt.h>
+#include <string.h>
+
+struct subcmd {
+	char *name;
+	char *doc;
+	enum swupd_code (*mainfunc)(int, char **);
+};
+
+static struct subcmd commands[] = {
+	{ "add", "Add third party repo", third_party_add },
+	{ "remove", "Remove third party repo", third_party_remove },
+	{ "list", "List third party repo", third_party_list },
+	{ 0 }
+};
+
+static int subcmd_index(char *arg)
```

Comment:
Instead of copying, export this function in swupd_internal.h and use it in 3rd_party

Reply:
@otaviobp This command is slightly diffferent than the one we used in main.c since we dont have to check if it is a verify command etc. Should I still use that same function? Also I will need to pass commands args to it since commands are different from main.c, add.c remove.c

Reply:
take a look at  #1180
And you can get that as a parameters. You can rename that if you find a better name and add a comment explaining what the function is doing.

Reply:
Okay Cool I will move to swupd_internal.h as a inline and include it in main.c and 3rd-party.c

Reply:
Makes sense will change in next push